echopype/echodata/zarr_combine.py

```diff
@@ -656,6 +657,13 @@ def _append_ds_list_to_zarr(
             for uniform_ind, non_uniform_dict in chunk_mapping.items():
                 for ds_list_ind, dim_slice in non_uniform_dict.items():
                     # get ds containing only variables who have dim in their dims
+                    cur_ds = ds_list[ds_list_ind]
+                    # get all variable names from dataset
+                    ds_keys = [k for k in cur_ds.variables.keys()]
+                    # compare variable names from dataset with the first
+                    # file's variable names and only grab the ones that exists
+                    # in the current dataset
+                    drop_names = [name for name in drop_names if name in ds_keys]
```

Comment:
here `drop_names` will change as the loop goes on. If the intention is to compare with the first file's variable names, maybe preserve that as a separate variable outside of the loop for use here?

Reply:
this may not be obvious if the list of eds to combine has only 2 elements

Reply:
Thanks for catching this!